image_substitutors_test.go

```diff
@@ -3,8 +3,51 @@ package testcontainers
 import (
 	"context"
 	"testing"
+
+	"github.com/testcontainers/testcontainers-go/internal/config"
 )
 
+func TestCustomHubSubstitutor(t *testing.T) {
+	t.Run("should substitute the image with the provided one", func(t *testing.T) {
+		s := NewCustomHubSubstitutor("quay.io")
+
+		img, err := s.Substitute("keycloak/keycloak:24.0")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if img != "quay.io/keycloak/keycloak:24.0" {
+			t.Errorf("expected quay.io/keycloak/keycloak:24.0, got %s", img)
+		}
+	})
+	t.Run("should not substitute the image if it is already using the provided hub", func(t *testing.T) {
+		s := NewCustomHubSubstitutor("quay.io")
+
+		img, err := s.Substitute("quay.io/keycloak/keycloak:24.0")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if img != "quay.io/keycloak/keycloak:24.0" {
+			t.Errorf("expected quay.io/keycloak/keycloak:24.0, got %s", img)
+		}
+	})
+	t.Run("should not substitute the image if hub image name prefix config exist", func(t *testing.T) {
+		config.Reset()
+		t.Setenv("TESTCONTAINERS_HUB_IMAGE_NAME_PREFIX", "registry.mycompany.com/mirror")
```

Comment:
Any test involving the config suffers from side effect (on my TODO to fix that), so I think we need to 1) put a config.Reset in `TestSubstituteBuiltImage` 2) call reset in a defer/t.Cleanup for the methods updating the config

Reply:
Fixed. Check it again please.